packages/french-verbs/src/index.ts

```diff
@@ -173,33 +228,77 @@ export function getAux(verb: string, aux: FrenchAux, pronominal: boolean | undef
     }
   }
 }
-function getConjugatedPasseComposePlusQueParfait(
+
+export function getAuxPassive(tense: Tense): FrenchAux {
+  if (isComposedTense(tense)) {
+    return 'AVOIR';
+  }
+  return 'ETRE';
+}
+
+export function getTenseAux(voice: Voice, tense: Tense): VerbInfoIndex {
+  if (voice === 'Pass') {
+    if (isComposedTense(tense)) {
+      return getTenseAux('Act', tense);
+    }
+    return tenseMapping[tense];
+  }
+  if (isComposedTense(tense)) {
+    switch (tense) {
+      case 'PASSE_COMPOSE':
+        return 'P';
+      case 'PLUS_QUE_PARFAIT':
+        return 'I';
+      case 'CONDITIONNEL_PASSE_1':
+        return 'C';
+      case 'CONDITIONNEL_PASSE_2':
+        return 'T';
+      case 'SUBJONCTIF_PASSE':
+        return 'S';
+      case 'INFINITIF_PASSE':
+        return 'W';
+      case 'PARTICIPE_PASSE_COMPOSE':
+        return 'G';
+      case 'PARTICIPE_PASSE': //not really useful
+        return 'P';
+    }
+  }
+
+  const err = new Error();
+  err.name = 'TypeError';
+  err.message = `The tense is not supported`;
+  throw err;
+}
+
+function getConjugatedComposedTenseOrPassive(
   verbInfo: VerbInfo,
   verb: string,
-  tense: string,
+  tense: Tense,
   person: number,
   composedTenseOptions: ComposedTenseOptions,
   pronominal: boolean,
   negativeAdverb: string | undefined,
   modifierAdverb: string | undefined,
+  voice: Voice,
 ): string {
   if (!composedTenseOptions) {
     const err = new Error();
     err.name = 'TypeError';
-    err.message = `ComposedTenseOptions is mandatory when tense is PASSE_COMPOSE or PLUS_QUE_PARFAIT`;
+    err.message = `ComposedTenseOptions is mandatory when tense is composed or voice passive`;

```

Comment:
Do you have a test case that triggers this exception?

Reply:
This was undetected unreachable code. I refactored to remove it.